src/server/pfs/reduce.go

```diff
@@ -16,6 +16,7 @@ func ReduceRepoInfos(repoInfos []*pfs.RepoInfo) []*pfs.RepoInfo {
 			continue
 		}
 		reducedRepoInfo.SizeBytes += repoInfo.SizeBytes
+		reducedRepoInfo.Provenance = repoInfo.Provenance
```

Comment:
It is an invariant. Asserting it gets a bit tricky though. For one reduce functions don't return errors so we'd have to add that and handle them. For a second there's no guarantee about the order of the `Provenance` although it should in theory be the same. We'd probably want to compare them as sets. Considering that we're going to be getting rid of this code when we do the pfs refactor my vote would be to ignore for now. I can implement it though if that makes you uneasy.